gzip data between volunteers and the server

omerjerk opened the pull request:
Signed-off-by: Umair Khan <omerjerk@gmail.com>

omerjerk commented:
Is this test failing because of my change?

elavoie commented on src/processor.js:
```diff
@@ -106,6 +107,7 @@ function createProcessor (node, opts) {
           }
         }),
         pull.map(function (x) { return String(x) }),
+        pull.map(x => zlib.gzipSync(x)),
```
Please use a regular anonymous function (e.g. 'function (x) { return x }') rather than an arrow function. That way, pando will work on older versions of Node without additional tooling such as babel.

elavoie commented on bin/index.js:
```diff
@@ -167,7 +168,7 @@ bundle(args.module, function (err, bundlePath) {
         probe('pando:result'),
         pull.through(log),
-        pull.through(function (x) { process.stdout.write(String(x) + '\n') }),
+        pull.through(function (x) { process.stdout.write(zlib.gunzipSync(x) + '\n') }),
```
This is where the problem happens. While this should work when pando is used as a commandline tool, this will not work when the processor is used as a library in another application. This is the use case in the test which fails in the continuous integration. I would suggest you decompress between line 454-455 in src/processor.js instead. The design goal for the processor abstraction is to be a drop-in replacement for pull.asyncMap but with the computation being distributed on different cores and machines.

The bin/index.js file should not contain logic that depends on the internals of the processor implementation (and therefore break abstraction), it should only contain the logic necessary to connect a processor with the command-line interface and Unix pipes.

You can run the same test as for the continuous integration on your machine with 'npm test'.

omerjerk replied:
Hi Erick,

I have understood very well why we shouldn't implement this logic inside the bin/index.js file. I am just trying to get a better idea of the source code before I send another commit.

omerjerk commented:
Hi @elavoie,

I have addressed the issues which you mentioned. Please have a look.

One thing is that, I wasn't completely sure if I should have used pull.map or pull.through here - https://github.com/elavoie/pando-computing/pull/4/files#diff-66203dd7a24695d85414f35f3a4fc49cR455

I used pull.map because that made sense to me, however, what would have been the complication of going with pull.through?

elavoie commented:
pull.map was the right thing to do. pull.through does not modify the underlying stream, it only pulls each value one at a time to inspect or log for example.
